scripts/genetic_backtester/darwin.js

```diff
@@ -560,44 +596,51 @@ let simulateGeneration = () => {
     let fieldsGeneral = ['selector', 'fitness', 'vsBuyHold', 'wlRatio', 'frequency', 'strategy', 'order_type', 'endBalance', 'buyHold', 'wins', 'losses', 'period', 'min_periods', 'days', 'params'];
     let fieldNamesGeneral = ['Selector', 'Fitness', 'VS Buy Hold (%)', 'Win/Loss Ratio', '# Trades/Day', 'Strategy', 'Order Type', 'Ending Balance ($)', 'Buy Hold ($)', '# Wins', '# Losses', 'Period', 'Min Periods', '# Days', 'Full Parameters'];
 
-    let csv = json2csv({
+    let dataCSV = json2csv({
       data: results,
       fields: fieldsGeneral,
       fieldNames: fieldNamesGeneral
     });
 
     let fileDate = Math.round(+new Date() / 1000);
-    let fileName = `simulations/backtesting_${fileDate}.csv`;
-    fs.writeFile(fileName, csv, err => {
-      if (err) throw err;
-    });
-
-    // let fileNameJSON = `simulations/backtesting_${fileDate}.json`;
-    // fs.writeFile(fileNameJSON, JSON.stringify(results, null, 2), err => {
-    //   if (err) throw err;
-    // });
-
+    let csvFileName = `simulations/backtesting_${fileDate}.csv`;
+    
     let poolData = {};
     selectedStrategies.forEach(function(v) {
       poolData[v] = pools[v]['pool'].population();
     });
 
-    let poolFileName = `simulations/generation_data_${fileDate}_gen_${generationCount}.json`;
-    let poolDataJSON = JSON.stringify(poolData, null, 2);
-    fs.writeFile(poolFileName, poolDataJSON, err => {
-      if (err) throw err;
+    let jsonFileName = `simulations/generation_data_${fileDate}_gen_${generationCount}.json`;
+    let dataJSON = JSON.stringify(poolData, null, 2);
+    var filesSaved = 0;
+    saveGenerationData(csvFileName, jsonFileName, dataCSV, dataJSON, (id)=>{
+      filesSaved++;
+      if(filesSaved == 2){        
+        console.log(`\n\nGeneration's Best Results`);
+        selectedStrategies.forEach((v)=> {
+          let best = pools[v]['pool'].best();      
+          if(best.sim)
+            console.log(`\t(${v}) Sim Fitness ${best.sim.fitness}, VS Buy and Hold: ${best.sim.vsBuyHold} End Balance: ${best.sim.endBalance}, Wins/Losses ${best.sim.wins}/${best.sim.losses}.`);
+          else 
+            console.log(`\t(${v}) Result Fitness ${results[0].vsBuyHold}, VS Buy and Hold: ${results[0].vsBuyHold}, End Balance: ${results[0].endBalance}, Wins/Losses ${results[0].wins}/${results[0].losses}.`);
```

Comment:
```
Result Fitness ${results[0].vsBuyHold}, VS Buy and Hold: ${results[0].vsBuyHold}
```
should be
```
Result Fitness ${results[0].fitness}, VS Buy and Hold: ${results[0].vsBuyHold}
```

Reply:
Good catch! Can't believe I missed that...